src/insns/cbo_exceptions.adoc

```diff
@@ -9,6 +9,11 @@ CBO.INVAL performs cache block flushes instead of
 invalidations for less privileged modes. The instruction checks shown in the
 table below remain unchanged regardless of the value of CBIE and the privilege
 mode.
+
+NOTE: A valid implementation of CBO.INVAL is to _always_ execute it as CBO.FLUSH.
+ This is recommended for CHERI systems as invalidate _could_ reveal stale capabilities
```

Comment:
I don't know if we want to go that far. Invalidate still leaks secrets in non-CHERI. I'd just want to highlight the potential to reveal capabilities from past memory contents that you wouldn't otherwise have access to as justification for why it needs ASR + write perms. INVAL is already FLUSH for userspace normally, per the CSR field that lets you enforce that.

Reply:
ok - so do you think we need this note at all? Or feel free to suggest a modification which weakens it.

Reply:
How about:

```NOTE: Invalidating a cache block can re-expose capabilities previously stored to it after the most recent flush, not just secret values. As such, CBO.INVAL has stricter checks on its use than CBO.FLUSH, and should only be made available to, and used by, sufficiently-trusted software. Untrusted software should use CBO.FLUSH instead.```

Reply:
I like that suggested wording.